packages/transition-common/src/services/batchCalculation/types.ts

```diff
@@ -1,26 +1,53 @@
 import { RoutingOrTransitMode } from 'chaire-lib-common/lib/config/routingModes';
+import { ErrorMessage } from 'chaire-lib-common/lib/utils/TrError';
 import {
     TransitRoutingQueryAttributes,
     validateTrQueryAttributes
 } from '../transitRouting/TransitRoutingQueryAttributes';
+import { _isBlank } from 'chaire-lib-common/lib/utils/LodashExtensions';
 
 export const isBatchParametersValid = (parameters: BatchCalculationParameters) => {
     let parametersValid = true;
     const errors: string[] = [];
-    if (parameters.routingModes.length === 0) {
+    if (!Array.isArray(parameters.routingModes) || parameters.routingModes.length === 0) {
         parametersValid = false;
         errors.push('transit:transitRouting:errors:RoutingModesIsEmpty');
-    }
-    if (parameters.routingModes.includes('transit')) {
+    } else if (parameters.routingModes.includes('transit')) {
         const { valid: queryAttrValid, errors: queryAttrErrors } = validateTrQueryAttributes(parameters);
         if (!queryAttrValid) {
             parametersValid = false;
             errors.push(...queryAttrErrors);
         }
     }
+    if (typeof parameters.cpuCount !== 'number' && typeof parameters.maxCpuCount === 'number') {
```

Comment:
Shouldn't this be moved to an external function, since it is used at multiple places?

Reply:
OK, but for some users like me, I would like to keep the CPUCount parameter available.